src/HalfSpace.jl

```diff
@@ -116,3 +116,33 @@ We just check if ``x`` satisfies ``a⋅x ≤ b``.
 function ∈(x::AbstractVector{N}, hs::HalfSpace{N})::Bool where {N<:Real}
     return dot(x, hs.a) <= hs.b
 end
+
+"""
+    halfspace_left(p::AbstractVector{N},
+                   q::AbstractVector{N})::HalfSpace{N} where {N<:Real}
+
+Return a half-space describing the 'left' of a line segment through two points.
+
+### Input
+
+- `p` -- first point
+- `q` -- second point
+
+### Output
+
+The half-space whose boundary goes through the two points `p` and `q` and which
+describes the left-hand side of the directed line segment `pq`.
+
+### Algorithm
+
+The implementation is simple: the half-space ``a⋅x ≤ b`` is calculated as
+`a = [dy, -dx]`, where ``d = (dx, dy)`` denotes the line segment
+`pq`, that is, ``\\vec{d} = \\vec{p} - \\vec{q}``, and `b = dot(p, a)` because
+`p` is on this line (we can equally choose `q` instead of `p`). 
+"""
+function halfspace_left(p::AbstractVector{N},
+                        q::AbstractVector{N})::HalfSpace{N} where {N<:Real}
+    @assert p != q "the points must not be equal"
```

Comment:
this notion of "left", i think, only makes sense in 2D, but `HalfSpace` is general so i think we should make this check here and throw an error otherwise. i can add it and see if you agree.

Reply:
1. I do not remember if I just chose the name from the 2D view or if I did not think about the nD case. But now that I do, the function is general, right? So maybe one day we want to use it in nD. Should we just find a better name?
2. Do you think we should also add a version from a `LineSegment` (which is just the one-liner which unwraps the two vertices and calls this function here)?